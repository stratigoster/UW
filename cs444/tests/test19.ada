-- no errors here
PACKAGE main IS
  TYPE Color IS (Red,Orange,Green,Blue,Indigo,Violet,White,Black);
  TYPE Light IS (Black,Neon,White,Red);
  TYPE Beam IS array (Color, Light) of Light;
  a_beam : Beam;
  x_ray : Light := Red;
  is_photon : Boolean := x_ray = White OR x_ray = Black;

END main;

PACKAGE BODY main IS
BEGIN
  null;
END main;
