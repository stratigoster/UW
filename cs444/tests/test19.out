

#!/usr/bin/env python
import math
def _forwardGen(seq):
	for i in seq:
		yield { 'all':i }
def _reverseGen(seq):
	i = len(seq)
	while i>0:
		i = i - 1
		yield { 'all':seq[i] }
class _dummy_(Exception):pass
class _numeric_error(Exception):pass
class _constraint_error(Exception):pass
class main:
	try:
		a_beam = { 'all':0 }
		x_ray = { 'all':3 }
		is_photon = { 'all':((x_ray['all'] == 2) | (x_ray['all'] == 0)) }
		def __init__(self):
			try:
				pass
				pass
			except _dummy_:
				pass
	except _dummy_:
		pass
main()
